buildSrc/src/main/groovy/org/elasticsearch/gradle/test/NodeInfo.groovy

```diff
@@ -165,24 +166,14 @@ class NodeInfo {
         }
 
         final String javaHome
-        final Map<Integer, JavaVersion> javaVersions = project.javaVersions
         if (nodeVersion.before("6.2.0")) {
-            final String java8Home = javaVersions.get(8)
-            if (java8Home == null) {
-                throw new GradleException("JAVA8_HOME must be set to run BWC tests against [" + nodeVersion + "]")
-            }
-            javaHome = java8Home
+            env = ['JAVA_HOME':"${-> getJavaHome(project, 8, "JAVA8_HOME must be set to run BWC tests against [" + nodeVersion + "]")}"]
         } else if (nodeVersion.onOrAfter("6.2.0") && nodeVersion.before("6.3.0")) {
-            final String java9Home = javaVersions.get(9)
-            if (java9Home == null) {
-                throw new GradleException("JAVA9_HOME must be set to run BWC tests against [" + nodeVersion + "]")
-            }
-            javaHome = java9Home
+            env = ['JAVA_HOME':"${-> getJavaHome(project, 9, "JAVA9_HOME must be set to run BWC tests against [" + nodeVersion + "]")}"]
         } else {
-            javaHome = project.compilerJavaHome
+            env = ['JAVA_HOME':project.runtimeJavaHome]
```

Comment:
This is a mistake from #29493; this should have remained as `RUNTIME_JAVA_HOME`.